proto/api/registration/registration.proto

```diff
@@ -85,6 +85,25 @@ message Bundle {
     common.Bundle bundle = 2;
 }
 
+// Represents a ListAgents response
+message ListAgentsResponse {
+    // List of all attested agents
+    repeated spire.common.AttestedNode nodes = 1;
+}
+
+// Represents an evict request
+message EvictAgentRequest {
+    // Core identity of the agent to be de-attested.
+    // For example: "spiffe://example.org/spire/agent/join_token/feea6adc-3254-4052-9a18-5eeb74bf214f"
+    string spiffeID = 1;
+}
+
+// Represents an evict response
+message EvictAgentResponse {
+    // If the entry is successfully deleted, deleteSucceed will be true
+    bool deleteSucceed = 1;
```

Comment:
I think we should avoid this pattern, it can be error prone because consumers may forget to check the value of `deleteSucceed`. Instead, if the delete has failed for some reason, we should return an error directly.

Instead of returning deleteSucceed, one option is to return spire.common.AttestedNode representing the entry we just deleted

Reply:
Good point! I'll modify the API to return the deleted node